### ISSUE TITLE
Better ergonomics in the command line

### ISSUE THREAD

**Issue by jonaprieto** [anoma/juvix#948]
This issue is about the ergonomics of the juvix command line. At the moment, one needs to provide a command
like parse/typecheck, the file, and the backend. For example, one possible scenario of using Juvix for the first time is the following. 

```
$ touch Temp.ju
$ juvix Temp.ju
Invalid argument `Temp.ju'

Usage: juvix COMMAND [-c|--config PATH]
```

Ok. Let us check the help command:

```
$ juvix --help
Juvix version 0.0.0-0db89d4

Usage: juvix COMMAND [-c|--config PATH]

Available options:
  -c,--config PATH         Path to YAML configuration
                           file (default: "/Users/jonaprieto/work/examples-juvix/juvix.yaml")
  -h,--help                Show this help text

Available commands:
  version                  Display version information
  config                   Adjust runtime configuration or generate an example
                           config file
  parse                    Parse a Juvix source file
  typecheck                Typecheck a Juvix source file
  compile                  Compile a Juvix source file
```

Now, we know what to do.

```
$ juvix parse Temp.ju
Missing: (-b|--backend BACKEND)
Usage: juvix parse INPUTFILE (-b|--backend BACKEND)
  Parse a Juvix source file
```

BACKEND? what are the options? At the moment, there is no way for you to know, unless check the source code.
The following could be also a possible attempt that fails.

```
$  juvix parse --help
Invalid option `--help'
...
```

Let us assume, we solved the issue above and then runs again, this time to get an error (reported in  #946).

```
$ juvix parse Temp.ju -b michelson
juvix: user error ("1:1:
```

In conclusion, we can do this better. However some choices have to be made. 

A few thoughts on this from my side.

- The current commands can be replaced by optional flags. For example,

```
$ juvix Temp.ju [--parse|--check] [-b|--backend] 
```
- There must be defaults for the beginner and options for the skilled user.

For example, the expected behavior of running the following command:

```
$ juvix Temp.ju
```

can be:
```
$ juvix Temp.ju --check -b michelson
```

- Also, we can have contextual command helpers (i.e. `juvix parse --help`).
